### ISSUE TITLE
UP034 false negative (not detected)

### ISSUE THREAD

**Issue by spaceone** [charliermarsh/ruff#2389]
not detected:
`ruff --isolated --select UP034 - <<<"set(('1.3.6.1.4.1.4203.666.11.1.4.2.12.1'),)"`

expected:
`-:1:5: UP034 Avoid extraneous parentheses`

**Comment by charliermarsh** [comment]
(`pyupgrade` doesn't catch it either.)

**Comment by spaceone** [comment]
https://github.com/asottile/pyupgrade/issues/791

**Comment by wookie184** [comment]
The behaviour wanted here is probably that of pylint's [`C0325`](https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/superfluous-parens.html). On https://github.com/charliermarsh/ruff/issues/970 `C0325` is marked as completed due to `UP034`, but the scope of `UP034` is much smaller so that doesn't seem correct.

The `UP043` rule could be expanded to cover more cases, but it's not really related to pyupgrade's purpose, so IMO it would make sense just to implement the pylint rule separately.

**Comment by Avasam** [comment]
Any single statement with an operator is also not detected:
ie: `test = (1) + 2`

**Comment by codereport** [comment]
Honestly, ruff `UP034` doesn't catch anything for me. pylint catches all of these:

```py
assert (a == b)
if (a == b):
while (a == b):
```
and returns
```
C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
C0325: Unnecessary parens after 'while' keyword (superfluous-parens)
C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
```

ruff catches nothing.

**Comment by charliermarsh** [comment]
Yeah I think they're just very different rules. I would describe UP034 as catching "redundant" parentheses, so, e.g., it will catch `((1)) + 2` but not `(1) + 2`. C0325 seems to catch "unnecessary" parentheses, i.e., all cases in which parentheses are not syntactically necessary, even if they may be used intentionally for readability or otherwise.

**Comment by codereport** [comment]
@charliermarsh Looking forward to when it is implemented. Currently the only thing I use pylint for is that check. Once it is added I can remove that dependency : ) I am pretty sure a `--fix` is possible as well and pylint doesn't do that. So it will be not only a replacement but an improvement : )

Thanks for all your work on this tool btw. I have never looked back since discovering it. I think I first heard about it on Talk Python to Me.

**Comment by ilyakamens** [comment]
Any idea when this will be added?

**Comment by pelson** [comment]
Came here to say that I was expecting `UP034` to fix unnecessary parentheses such as the one in the example below:

```
$ echo 'import os; str( u"", errors=("strict"))' > foo.py; ruff check --fix --extend-select UP034 ./foo.py; cat foo.py | ruff format -
Found 1 error (1 fixed, 0 remaining).
str("", errors=("strict"))
```

(and also wish I was able to pipe the `ruff check --fix` to make the reproducer more elegant 😉, #20460)

**Comment by spaceone** [comment]
> (and also wish I was able to pipe the ruff check --fix to make the reproducer more elegant 😉, https://github.com/astral-sh/ruff/issues/20460)

You can use bash for this:

```
$ ruff check --isolated --diff --fix --extend-select UP034 <(echo 'import os; str( u"", errors=("strict"))')
--- /dev/fd/63
+++ /dev/fd/63
@@ -1 +1 @@
-import os; str( u"", errors=("strict"))
+str( u"", errors=("strict"))

Would fix 1 error.

```
